phpunit/blocks/render-block-heading-test.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Navigation block rendering tests.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ */
+
+/**
+ * Tests for the Navigation block.
+ *
+ * @group blocks
+ */
+class Render_Block_Heading_Test extends WP_UnitTestCase {
+
+	/**
+	 * @covers ::gutenberg_block_core_heading_render
+	 * @dataProvider add_css_class_test_examples
+	 */
+	public function test_block_core_heading_render_appends_css_class_to_a_vanilla_element( $input, $expected_result ) {
+		$actual   = gutenberg_block_core_heading_render( array(), $input );
+		$this->assertEquals( $expected_result, $actual );
+	}
+
+	public function add_css_class_test_examples(  ) {
```

Comment:
I ❤️ the data provider. Can we add array keys, so it easier to debug on test failure. See this example.

https://github.com/WordPress/wordpress-develop/blob/2648a5f984b8abf06872151898e3a61d3458a628/tests/phpunit/tests/rest-api/application-passwords.php#L114-L124